### PR TITLE
Improve test performance

### PR DESCRIPTION
**Pull request by erikh**
The volume CRUD primitives were lousy. I moved on @mapuri's randomized
containers patches and did the same for volumes. Because of a bug in docker
(see below) we cannot create the same volume twice in a short period of time
without hitting a cache bug:

docker/docker#24205 for more information.

This fix implements a cleaner API for managing volumes, and converts the whole
suite to use those features, including the randomized volume names.

### CODE REVIEW

**Comment by erikh** [comment]
This patch improves test performance by over 200%

File: systemtests/util_test.go [hunk 1]
```diff
@@ -22,6 +22,44 @@ var startedContainers struct {
...
+func volumeParts(volume string) (string, string) {
+	parts := strings.SplitN(volume, "/", 2)
+	return parts[0], parts[1] // panic, schmanick
+}
+
+func fqVolume(policy, volume string) string {
+	return strings.Join([]string{policy, volume}, "/")
+}
+
+func genRandomVolume() string {
```
Comment (unclejack) [inline systemtests/util_test.go]: Could this run into collisions because genRandomString doesn't avoid collisions?

Comment (erikh) [inline systemtests/util_test.go]: the next 10 lines of code are collision detection and retry.

Comment (unclejack) [inline systemtests/util_test.go]: Ok, got it.

**Comment by unclejack** [comment]
LGTM

**Comment by dseevr** [comment]
LGTM

**Comment by yuva29** [comment]
LGTM

**Pull request merged**